src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -599,6 +639,25 @@ private BoundExpression CreateAndPopulateArray(BoundCollectionExpression node, A
                             _factory.Binary(BinaryOperatorKind.Addition, indexTemp.Type, indexTemp, _factory.Literal(1)),
                             isRef: false,
                             indexTemp.Type));
+                },
+                (sideEffects, targetArray, spreadElement, rewrittenSpreadOperand) =>
+                {
+                    if (TryPrepareCopyToOptimization(spreadElement, rewrittenSpreadOperand) is not var (spanSliceMethod, spreadElementAsSpan, getLengthMethod, copyToMethod))
+                        return false;
+
+                    if (targetSpanTemp is null)
+                    {
+                        var targetSpan = TryConvertToSpanOrReadOnlySpan(targetArray);
```

Comment:
It could be, but I wasn't able to cook up a scenario that outright breaks things. I am going to try microbenchmarking doing repeated conversions to Span versus using a Span temp.

Reply:
My conclusion on this is:
- Yes, overhead is minimal. Probably on the order of 2% execution time for collection-exprs we are likely to see.
- I believe that the impl is currently "await-safe". All the spread elements have been evaluated to temps at the time we take a span-temp of the destination. So we won't visit any awaits while copying the spreads.
- However, this assumption is fragile, and the failure mode of that assumption breaking later could be bad.
- So, for simplicity, let's eliminate the dest temp for now, and open an issue to reintroduce it later if we want.